README.md

```diff
@@ -27,6 +29,24 @@ module "example" {
 }
 ```
 
+### Single Account Usage ###
+
+```hcl
+module "example" {
+  source = "github.com/cisagov/molecule-iam-user-tf-module"
+
+  aws_region = "us-east-2"
```

Comment:
I don't think this single account example is correct.  Unless I'm mistaken, I think it still needs a providers block, but it would look like this:
```
providers = {
  aws                   = aws
  aws.images-production = aws
  aws.images-staging    = aws
}
```

Also, I think showing that we can override the default value of the `aws_region` variable here might be confusing, since we don't show it in the "multi-account" example.  When I look at this, it implies that doing this has something to do with the fact that this is a single-account example, even though it does not.

Reply:
If we're going to expect using the module to include passing the providers explicitly, then I don't think we should have a module variable for the AWS region. The variable only serves a purpose if we are modifying the providers the module uses by default. I think having a variable that does nothing if following our examples is equally confusing. We could switch to a local variable in `providers.tf` to ensure that the default is consistent. @jsf9k what are your thoughts?

Reply:
Here's what is confusing me- and maybe I'm misunderstanding something here.  If we allow the providers to be inherited from the module caller, like so:
```
module "example" {
  source = "github.com/cisagov/molecule-iam-user-tf-module"

  aws_region = "us-east-2"

  ssm_parameters = ["/example/parameter1", "/example/config/*"]
  user_name      = "test-molecule-iam-user-tf-module"

  tags = {
    Team        = "VM Fusion - Development"
    Application = "molecule-iam-user-tf-module testing"
  }
}
```

Then those providers still need to be defined somewhere by the caller, which looks something like this (lifted from the example):
```
# Default AWS provider (ProvisionAccount for the Users account)
provider "aws" {
  region  = "us-east-1"
  profile = "cool-users-provisionaccount"
}

# ProvisionParameterStoreReadRoles AWS provider for the
# Images (Production) account
provider "aws" {
  region  = "us-east-1"
  profile = "cool-images-production-provisionparameterstorereadroles"
  alias   = "images-production"
}

# ProvisionParameterStoreReadRoles AWS provider for the
# Images (Staging) account
provider "aws" {
  region  = "us-east-1"
  profile = "cool-images-staging-provisionparameterstorereadroles"
  alias   = "images-staging"
}
```

The `region` is required in those providers defined by the caller.  So even if we define a variable for the region and pass that into the module, the module is still going to inherit the providers (and corresponding regions) defined by the caller, right?

If all that is correct, then it means that there is no point in passing a variable into the module for the region.  If something about that is wrong, please let me know where.

BTW, I wanted to test this out locally to verify if the module's provider region (i.e. the one set by the variable) can override the provider region inherited from the caller, but I don't think I can in this case since all of the resources that get created are global (IAM user and roles)- I would have no way of knowing which region was actually used.

Reply:
This all started because these Terraform modules are starting to fail the terraform-validate pre-commit hook.  It sounds like we just need to put _something_ there for the region in order to get that hook to pass, and that something will always be ignored.  I'm fine with that being a local variable, or not a variable at all.

@mcdonnnj, do you want to revert ad358f1?

Reply:
So this is a more complete discussion let me background what's happening. These modules are not starting to fail the `terraform validate` pre-commit hook. In June of last year the `pre-commit-terraform` hook did some Terraform v0.12 updates. As seen [on this line](https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71#diff-0560dc7f5fcce3201df9ddfd7eef7d1fL37), the hook we were using, `terraform_validate_no_variables`, was combined and renamed to `terraform_validate`. It wasn't  caught until last month (March of 2020), and the hood ID being used was fixed in https://github.com/cisagov/skeleton-generic/pull/40. So I believe this would have been failing for us ever since Terraform v0.12 changes, we just didn't know because of a misconfiguration.

@dav3r 
Reading from [Terraform documentation](https://www.terraform.io/docs/configuration/modules.html#passing-providers-explicitly) it looks like the way you provided is how we should show (and do) single account usage. I haven't really seen much alias usage before so I was not well read on the particulars that introduces.

I have two possibilities in mind to resolve this:
1. We use a local variable in `providers.tf` to set the region and thus it will pass, even if the region is somewhat superfluous.
2. Everyone can make sure they have an `AWS_DEFAULT_REGION` environment value set. A corresponding variable would be configured for the GitHub Actions workflow to provide the same functionality (I have tested this locally after removing the `region` declarations in `providers.tf` for the module).

I'm leaning toward the environment variable because that would be a blanket fix for this in the future instead of continuously needing to provide explicitly declared `region` arguments to the AWS provider.

@jsf9k I'd probably just do a fresh commit to resolve the variable situation. I would like to keep the single account example and just update it to reflect the usage @dav3r showed.

Reply:
Thanks for the great explanation.  I'm not sure if I'm understanding that quote correctly:
> ...or via a shared credentials file if profile is specified.
Is this referring to the `~/.aws/config` file:
```
[default]
region = us-east-1
```

Reply:
Yes, that's correct. Sorry I missed that in my hunt for sources for clear explanation. That option is explicitly laid out in the [AWS provider reference](https://www.terraform.io/docs/providers/aws/index.html).
> `shared_credentials_file` = (Optional) This is the path to the shared credentials file. If this is not set and a profile is specified, `~/.aws/credentials` will be used.

Reply:
After a team discussion we opted for the local variable option from the two options I brought forward. This can be seen implemented in 326483f.